feat: add wrong assignment report dialog and webhook for routing form bookings

PeerRich opened the pull request:
<img width="915" height="780" alt="image" src="https://github.com/user-attachments/assets/fff07a72-31cd-43a7-ba38-5b5b15875f31" />


## What does this PR do?

Adds the ability to report wrong assignments for bookings that came through routing forms. This includes:

- A new "Wrong Assignment" dialog that displays the routing reason, guest email (with copy-to-clipboard), and host information
- Making the routing reason badge clickable to open this dialog
- Adding a "Report wrong assignment" option in the booking actions dropdown (with `user-x` icon) for routing form bookings
- A new `WRONG_ASSIGNMENT_REPORT` webhook that fires when the form is submitted, containing the booking data plus the report details
- Team member selector for "Who should have received it?" field (for team bookings)
- Mandatory "Additional notes" field with informational alert about CRM data quality
- Documentation for the webhook payload in `docs/developing/guides/automation/webhooks.mdx`

**Requested by:** @PeerRich (peer@cal.com)

**Link to Devin run:** https://app.devin.ai/sessions/36d28df5c5db408dbafce87b82e921e5

## Updates since last revision

- Resolved merge conflicts with main branch (conflicts in `BookingRepository.ts`)
- Added `WRONG_ASSIGNMENT_REPORT` webhook payload documentation to `docs/developing/guides/automation/webhooks.mdx`
- Moved webhook documentation into the Accordion section per review feedback (consistent with other triggers)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). Added webhook payload documentation.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Navigate to `/bookings` page
2. Find a booking that came through a routing form (should have a routing reason badge)
3. **Test badge click:** Click on the routing reason badge - the "Wrong Assignment" dialog should open
4. **Test dropdown action:** Click the three-dot menu on the booking, find "Report wrong assignment" option (should have `user-x` icon)
5. **Test dialog content:** Verify the dialog shows:
   - Routing reason (display-only)
   - "Who booked it?" with guest email and copy button
   - "Who received it?" with host name/email (display-only)
   - "Who should have received it?" - team member selector for team bookings, email input otherwise (optional)
   - "Additional notes" text area (mandatory)
   - Info alert about CRM data quality
6. **Test webhook:** Set up a webhook with the `WRONG_ASSIGNMENT_REPORT` trigger, submit the form, and verify the webhook is received

**Webhook payload structure:**
```json
{
  "booking": {
    "uid": "...",
    "id": 123,
    "title": "...",
    "startTime": "...",
    "endTime": "...",
    "status": "...",
    "eventType": { "id": 1, "title": "...", "slug": "...", "teamId": 1 }
  },
  "report": {
    "reportedBy": { "id": 1, "email": "...", "name": "..." },
    "routingReason": "...",
    "guest": "guest@example.com",
    "host": { "email": "host@example.com", "name": "Host Name" },
    "correctAssignee": "correct@example.com",
    "additionalNotes": "..."
  }
}
```

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked if my changes generate no new warnings

## Human Review Checklist

- [ ] Verify the dialog UI matches expected design
- [ ] Verify webhook payload structure is correct for the use case
- [ ] Test that the badge click and dropdown action both work correctly
- [ ] Verify the webhook fires for both team webhooks AND individual user webhooks (this is intentional per requirements)
- [ ] Verify the team member selector appears for team bookings and email input for non-team bookings
- [ ] Verify the new webhook documentation in `docs/developing/guides/automation/webhooks.mdx` is accurate and properly formatted in the Accordion

devin-ai-integration[bot] committed:
feat: add wrong assignment report dialog and webhook for routing form bookings

- Add WrongAssignmentDialog component showing routing reason, guest, and host
- Make routing reason badge clickable to open the dialog
- Add 'Report wrong assignment' option in action dropdown for routing form bookings
- Add WRONG_ASSIGNMENT_REPORT webhook trigger type
- Create tRPC endpoint to handle wrong assignment report and trigger team webhooks
- Add i18n translations for new UI strings

Co-Authored-By: peer@cal.com <peer@cal.com>

devin-ai-integration[bot] committed:
feat: extend wrong assignment webhook to support individual users

- Remove team-only restriction for wrong assignment reports
- Add userId parameter to getWebhooks call to support user webhooks
- Now triggers webhooks for both team and individual user bookings

Co-Authored-By: peer@cal.com <peer@cal.com>

devin-ai-integration[bot] committed:
feat: use AddMembersWithSwitch for team member selection in wrong assignment dialog

Co-Authored-By: peer@cal.com <peer@cal.com>

devin-ai-integration[bot] committed:
feat: make additional notes mandatory and add CRM info alert

Co-Authored-By: peer@cal.com <peer@cal.com>

devin-ai-integration[bot] committed:
feat: add WRONG_ASSIGNMENT_REPORT to webhook trigger options in UI

Co-Authored-By: peer@cal.com <peer@cal.com>

devin-ai-integration[bot] committed:
feat: update dialog labels and add copy-to-clipboard for guest email

Co-Authored-By: peer@cal.com <peer@cal.com>

devin-ai-integration[bot] committed:
feat: simplify team member selector to single select and add user-x icon

Co-Authored-By: peer@cal.com <peer@cal.com>

devin-ai-integration[bot] committed:
fix: remove duplicate select_team_member translation key

Co-Authored-By: peer@cal.com <peer@cal.com>

PeerRich committed:
Merge branch 'main' into devin/wrong-assignment-dialog-1765563324

devin-ai-integration[bot] committed:
Merge branch 'main' into devin/wrong-assignment-dialog-1765563324

Co-Authored-By: peer@cal.com <peer@cal.com>

devin-ai-integration[bot] committed:
fix: resolve biome lint warnings in WrongAssignmentDialog.tsx

Co-Authored-By: peer@cal.com <peer@cal.com>

devin-ai-integration[bot] committed:
fix: remove JSX from react import (use global namespace)

Co-Authored-By: peer@cal.com <peer@cal.com>

alishaz-polymath commented:
DevinAI this PR is introducing a new webhook trigger. Please add the respective payload shape (with example values) in the docs here: `docs/developing/guides/automation/webhooks.mdx` for this webhook trigger so that the docs are in sync with this new webhook trigger (WRONG_ASSIGNMENT_REPORT)

devin-ai-integration[bot] committed:
Merge branch 'main' into devin/wrong-assignment-dialog-1765563324

Co-Authored-By: ali@cal.com <ali@cal.com>

devin-ai-integration[bot] committed:
docs: add WRONG_ASSIGNMENT_REPORT webhook payload documentation

Co-Authored-By: ali@cal.com <ali@cal.com>

alishaz-polymath commented on docs/developing/guides/automation/webhooks.mdx:
```diff
@@ -1613,6 +1614,72 @@ These webhooks are triggered **automatically** when hosts or guests don't join a
+| report.host | Object | Current host details (email, name) |
+| report.correctAssignee | String | Email of who should have received the booking (optional) |
+| report.additionalNotes | String | Additional context provided by the reporter |
+
```
DevinAI let us instead place this in the Accordion above, like other triggers. That's much cleaner.

devin-ai-integration[bot] committed:
docs: move WRONG_ASSIGNMENT_REPORT to Accordion section per review feedback

Co-Authored-By: ali@cal.com <ali@cal.com>

Udit-takkar committed:
Merge branch 'main' into devin/wrong-assignment-dialog-1765563324

Udit-takkar committed:
feat: add webhook trigger

Udit-takkar committed:
fix: type error

Udit-takkar committed:
fix: type error

devin-ai-integration[bot] committed:
fix: restore routing reason display in WrongAssignmentDialog

Uncommented the routing reason section that was accidentally commented out.
This restores the intended dialog content that shows important context
for the wrong assignment report.

Addresses Cubic AI review feedback (confidence 9/10).

Co-Authored-By: unknown <>

devin-ai-integration[bot] committed:
Merge branch 'devin/wrong-assignment-dialog-1765563324' of https://git-manager.devin.ai/proxy/github.com/calcom/cal.com into devin/wrong-assignment-dialog-1765563324

Udit-takkar committed:
tests: add unit tests
